src/hott/06-contractible.rzk.md

```diff
@@ -553,3 +553,131 @@ together the two identifications to the out and back path.
+          ( x)
+          ( homotopy-contraction A is-contr-A x))
+        ( u)
+
+#def has-retraction-total-type-to-second-center-is-contr-base
+  ( A : U)
+  ( is-contr-A : is-contr A)
+  ( C : A → U)
+  : has-retraction
+    ( Σ ( x : A) , C x)
+    ( C (center-contraction A is-contr-A))
+    ( total-type-to-second-center-is-contr-base A is-contr-A C)
+  :=
+    ( ( second-center-to-total-type-is-contr-base A is-contr-A C)
+    , \ (x , u) →
+    ( rev (Σ (x : A) , C x)
+        ( x , u)
+        ( second-center-to-total-type-is-contr-base A is-contr-A C
+          ( total-type-to-second-center-is-contr-base A is-contr-A C (x , u)))
+        ( transport-lift A C
+            ( x)
+            ( center-contraction A is-contr-A)
+            ( rev
+              ( A)
+              ( center-contraction A is-contr-A)
+              ( x)
+              ( homotopy-contraction A is-contr-A x))
+            ( u))))
+
+#def has-section-total-type-to-second-center-is-contr-base
+  ( A : U)
+  ( is-contr-A : is-contr A)
+  ( C : A → U)
+  : has-section
+    ( Σ ( x : A) , C x)
+    ( C (center-contraction A is-contr-A))
+    ( total-type-to-second-center-is-contr-base A is-contr-A C)
+  :=
+    ( ( second-center-to-total-type-is-contr-base A is-contr-A C)
+    , \ u →
+      ( concat
+        ( C (center-contraction A is-contr-A))
+        ( total-type-to-second-center-is-contr-base A is-contr-A C
+          ( second-center-to-total-type-is-contr-base A is-contr-A C u))
+        ( transport A C
+          ( center-contraction A is-contr-A)
+          ( center-contraction A is-contr-A)
+          ( first-path-Σ A C (center-contraction A is-contr-A , u)
+            ( center-contraction A is-contr-A , u)
+            ( refl))
+          ( u))
+        ( u)
+        ( transport2
+          ( A)
+          ( C)
+          ( center-contraction A is-contr-A)
+          ( center-contraction A is-contr-A)
+          ( rev
+            ( A)
+            ( center-contraction A is-contr-A)
+            ( center-contraction A is-contr-A)
+            ( homotopy-contraction A is-contr-A
+              ( center-contraction A is-contr-A)))
+          ( refl)
+          ( all-paths-equal-is-contr
+            ( A)
+            ( is-contr-A)
+            ( center-contraction A is-contr-A)
+            ( center-contraction A is-contr-A)
+            ( rev
+              ( A)
+              ( center-contraction A is-contr-A)
+              ( center-contraction A is-contr-A)
+              ( homotopy-contraction A is-contr-A
+                ( center-contraction A is-contr-A)))
+              ( refl))
+            ( u))
+        ( second-path-Σ A C
+          ( ( center-contraction A is-contr-A) , u)
+          ( ( center-contraction A is-contr-A) , u)
+          ( refl))))
+
+#def equiv-total-type-second-center-is-contr-base
+  ( A : U)
+  ( is-contr-A : is-contr A)
+  ( C : A → U)
+  : Equiv
+    ( Σ ( x : A) , C x)
+    ( C (center-contraction A is-contr-A))
```

Comment:
I wonder, if it would make sense to prove a version of this for any point in A, rather than specifically the center of contraction?

Reply:
If A is contractible and then any two points are equal, so at that point it would be a simple transport?

Reply:
I was thinking why delegate the transport onto the user, but I see now that the user would likely choose the center of contraction anyway.

Reply:
I wrote a proof for that, I'll format and commit now.